File: src/main/java/um/haberes/report/model/haberes/core/CargoClasePeriodoDto.java
package um.haberes.report.model.haberes.core;

import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CargoClasePeriodoDto {

    private Long cargoClasePeriodoId;

    private Long legajoId;

    private Long cargoClaseId;

    private Integer dependenciaId;

    private Integer facultadId;

    private Integer geograficaId;

    private Long periodoDesde;

    private Long periodoHasta;

    private int horas = 0;

    private BigDecimal valorHora = BigDecimal.ZERO;

    private String descripcion;

    private PersonaDto persona;

    private CargoClaseDto cargoClase;

    private DependenciaDto dependencia;

    private FacultadDto facultad;

    private GeograficaDto geografica;

}

File: src/main/java/um/haberes/report/kotlin/dto/haberes/core/CargoClasePeriodoDto.kt
package um.haberes.report.kotlin.dto.haberes.core

import java.math.BigDecimal

data class CargoClasePeriodoDto(

    var cargoClasePeriodoId: Long? = null,
    var legajoId: Long? = null,
    var cargoClaseId: Long? = null,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var periodoDesde: Long? = null,
    var periodoHasta: Long? = null,
    var horas: Int = 0,
    var valorHora: BigDecimal = BigDecimal.ZERO,
    var descripcion: String? = null,
    var persona: PersonaDto? = null,
    var cargoClase: CargoClaseDto? = null,
    var dependencia: DependenciaDto? = null,
    var facultad: FacultadDto? = null,
    var geografica: GeograficaDto? = null

)
